README.md

```diff
@@ -1,2 +1,142 @@
-# cpp-gl
-General purpose template graph library for modern C++
+# CPP-GL
+
+General purpose header-only template graph library for C++20 and newer standards.
+
+[![g++](https://github.com/SpectraL519/cpp-gl/actions/workflows/gpp.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/g++)
+[![clang++](https://github.com/SpectraL519/cpp-gl/actions/workflows/clang.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/clang++)
+[![format](https://github.com/SpectraL519/cpp-gl/actions/workflows/format.yaml/badge.svg)](https://github.com/SpectraL519/cpp-gl/actions/workflows/format)
+
+<br />
+
+## Overview
+
+- The goal of the project was to create a higlhy customizable, intuitive and easy to work with graph library for the modern C++ standards.
+- The `CPP-GL` library's implementation relies solely on the C++ standard library (it does not require installing any additional tools) and is designed to be compatible with the C++ standard tools, e.g. range-based loops, std algorithms, the [ranges library](https://en.cppreference.com/w/cpp/ranges), stream operations, etc.
+- The library relies heavily on [concepts](https://en.cppreference.com/w/cpp/language/constraints) to acheive abstraction instead of interfaces and abstract classes to minimize the overhead associated with virtual tables and dynamic dispatch.
+
+> [!NOTE]
+> Versions `v1.0.*` of the library and the `v1.0` pre-release versions have been developed as the Bachelor of Engineering Thesis at the *Wrocław University of Science and Technology*
```

Comment:
Maybe:
- Versions `v1.0.*` and `v1.0-prerelease-*`
- Bachelor of Engineering Project